### PR TITLE
Add `otel_headers` to OTel logic

### PR DESCRIPTION
**Pull request by Kludex**
Does anyone know why just doing `export OTEL_EXPORTER_OTLP_HEADERS='Authorization=your-write-token'` on the terminal, before running airflow isn't enough? Airflow ignores environment variables?

### CODE REVIEW

File: airflow/config_templates/config.yml [hunk 1]
```diff
@@ -1239,6 +1246,13 @@ traces:
       type: string
       example: ~
       default: "8889"
+    otel_headers:
+      description: |
+        Headers to be added to the traces.
+      version_added: 2.10.0
```
Comment (Kludex) [inline airflow/config_templates/config.yml]: This is not true, I guess? 👀

Comment (jedcunningham) [reply airflow/config_templates/config.yml]: No, it'd be `3.0.0`.

File: airflow/config_templates/config.yml [hunk 2]
```diff
@@ -1174,6 +1174,13 @@ metrics:
...
```
Comment (Kludex) [inline airflow/config_templates/config.yml]:
```suggestion
      version_added: 3.0.0
```

File: airflow/config_templates/config.yml (hunk 1 above)
Comment (Kludex) [reply airflow/config_templates/config.yml]:
```suggestion
      version_added: 3.0.0
```

**Comment by Kludex** [comment]
I see an analogous PR that didn't have any tests: https://github.com/apache/airflow/pull/42441/

Can someone help me to move this forward? Also, the question no one is answering me: https://apache-airflow.slack.com/archives/C06K9Q5G2UA/p1732208803722989 😅

**Comment by Kludex** [comment]
This PR is not enough, and actually doesn't need to land yet. A bigger problem is that I can't not set a `port`.

I think instead of building the endpoint, we should allow set `otel_endpoint`, and make the other endpoints optional.

I just want to set: `https://logfire-api.pydantic.dev/v1/traces`.

**Comment by Kludex** [comment]
Since https://github.com/apache/airflow/pull/44346 was merged, we don't need this.

**Pull request closed without merging**